wikimedia/wikipedia-ios
# iOS 14 back button

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T258216

### Notes
* *What branch should this be merged into?* (It was branched off `xcode-12-beta`.) Are we going to make `main` xcode-12 ready? Should this PR be held until that is complete?
* I tested this on iOS 13, on three languages (including an RTL one), and on iPad. I don't believe there are any regressions, but I'd love some good poking at this to make sure there really are no regressions.
* Only tricky part left: For some reason, we're losing the "W" button titleView when jumping back from a long press. I spent quite a while, and couldn't figure out why. I could confirm it happens between `viewWillAppear` and `viewDidAppear`. I'm hackily now re-adding it on `viewDidAppear`, but it means there is a small flash when jumping back via the long press menu - it starts with no button, but then the button appears a split second after the user sees the article. Happy to keep working on the cause, if we want. (Among other things, I did look everywhere we set a `titleView` in the app, to see if something jumps out as being the cause.) 

### Test Steps
1. Load lots of articles. Ensure a long press on iOS 14 allows you to jump back anywhere. Repeat on different tabs (Explore, Search, etc.).
1. Try on iPad.
1. Try on iOS 12/13, to make sure there weren't any regressions. We'll see new back button titles (short article titles show as back button title, long article titles cause back button text to remain as `Back`).
1. Make sure it looks ok in another language.

### Screenshots/Videos *(if applicable)*
<img width="200" src="https://user-images.githubusercontent.com/9295855/88353199-6f14e580-cd11-11ea-8130-7b33638294ca.png"><img width="200" src="https://user-images.githubusercontent.com/9295855/88353206-73410300-cd11-11ea-84a2-df21651bdcae.png"><img width="200" src="https://user-images.githubusercontent.com/9295855/88353207-74723000-cd11-11ea-8a22-fc30b9d5332d.png"><img width="200" src="https://user-images.githubusercontent.com/9295855/88353209-75a35d00-cd11-11ea-8521-7166442c822d.png"><img width="200" src="https://user-images.githubusercontent.com/9295855/88353210-7936e400-cd11-11ea-99b8-4cf8b9debd5f.png"><img width="200" src="https://user-images.githubusercontent.com/9295855/88353219-7cca6b00-cd11-11ea-9824-2f9d267e06b7.png">
